backend/src/main/scala/cromwell/backend/standard/callcaching/RootWorkflowFileHashCacheActor.scala

```diff
@@ -95,7 +97,8 @@ class RootWorkflowFileHashCacheActor private(override val ioActor: ActorRef, wor
           case FileHashValueNotRequested =>
             log.error(s"Programmer error! Not expecting a hash request timeout when a hash value has not been requested: ${fileHashContext.file}")
           case v =>
-            log.error(s"Programmer error! Not expecting a hash request timeout when hash value '$v' is already in the cache: ${fileHashContext.file}")
+            log.info(s"Received hash request timeout when hash value '$v' was already in the cache: ${fileHashContext.file}." +
+              s"This is normal and happens due to race condition between scheduler thread responsible for firing timeouts and actor's message processing thread.")
```

Comment:
I just want to check my understanding is right:

Is this saying that the timer has already fired on the scheduler thread and the timer therefore can not be cancelled, but the timeout message may still arrive at the cache actor's mailbox after the hash value message. So is the race condition internal to the timer in that it doesn't guarantee that the message it sends arrives in the target mailbox before the timer becomes uncancellable?

Reply:
Not exactly "internal". The problem as I see it is that timer cancelation is done by `RootWorkflowFileHashCacheActor` as a part of work which it does while handling `IoSuccess`/`IoFailure` message. So it may happen that timer in scheduler thread fires when RWFHCA is already processing IO-message, but before it reached the line where it can cels the timer. In this case timeout message will successfully reach actor's message queue.

Reply:
but isn't the timer being cancelled first?

Reply:
It is, but since cancelation and firing timeout happen in different threads, the race condition is still possible.

Reply:
Visual representation:
![image](https://user-images.githubusercontent.com/4853242/73574645-38d26980-4444-11ea-8d7c-5bef656cd1cf.png)